src/internal.c

```diff
@@ -3658,7 +3658,8 @@ int RsaSign(WOLFSSL* ssl, const byte* in, word32 inSz, byte* out,
 #if defined(HAVE_PK_CALLBACKS)
     if (ssl->ctx->RsaSignCb) {
         void* ctx = wolfSSL_GetRsaSignCtx(ssl);
-        ret = ssl->ctx->RsaSignCb(ssl, in, inSz, out, outSz, keyBuf, keySz,
+        unsigned int tempOutSz = *outSz;
```

Comment:
This still needs to return `outSz`. Either add `*outSz = tempOutSz` after or go ahead and modify the RsaSignCb to use `word32` instead of `unsigned int` for this PR.